server/src/test/termdb.filter.unit.spec.js

```diff
@@ -120,8 +120,8 @@ tape('nested filter', async function (test) {
 	//console.log(filter.values)
 	test.deepEqual(
 		Object.keys(filter).sort((a, b) => (a < b ? -1 : 1)),
-		['CTEname', 'CTEs', 'filters', 'values'],
-		'should return an object with the four expected keys'
+		['CTEname', 'CTEs', 'filters', 'sampleTypes', 'values'],
+		'should return an object with the five expected keys'
 	)
```

Comment:
Since this is a big change that can affect the data request for many plots, we should add unit tests for bar chart with overlay sql results or server response to:

- a request for child-samples-only, without a filter, and then with filter 
- a request for parent-samples-only, without a filter, and then with filter
- a request with parent and child samples, with and without a filter
- the filter in the tests above can be varied to have a tvs term that's annotated on a parent sample, then child sample, then combination of these

Reply:
as long as other ds are not broken, i'm fine to do these tests in next branch, so that this pr can be merged soon

Reply:
The db generation now works also for the default dataset, in a next PR I will add the missing values in the sample view.